### PR TITLE
Change to use napps_dir from kytos instance

### PR DESCRIPTION
**Pull request by macartur**
kytos commands will request the configuration loaded by kytos instance
before install and enable napps.
Fix kytos/kytos-utils#85

### CODE REVIEW

File: kytos/utils/config.py [hunk 1]
```diff
@@ -82,22 +82,15 @@ def set_env_or_defaults(self):
...
-            if not napps_path:  # default paths
-                base = os.environ.get('VIRTUAL_ENV') or '/'
-                napps_path = os.path.join(base, 'var', 'lib', 'kytos', 'napps')
-            self.config.set('napps', 'enabled_path', napps_path)
-            self.config.set('napps', 'installed_path',
-                            os.path.join(napps_path, '.installed'))
+        if kytos_server is not None:
+            self.config.set('kytos', 'api', kytos_server)
+        elif not self.config.has_option('kytos', 'api'):
+            self.config.set('kytos', 'api', 'http://localhost:8181/')
```
Comment (diraol) [inline kytos/utils/config.py]: We can make this code more pythonic and more clear doing:
```python
from collections import namedtuple

Option = namedtuple('Option', ['section', 'name', 'env_var', 'default_value'])
options = [Option('auth', 'user', 'NAPPS_USER', None),
               Option('auth', 'token', 'NAPPS_TOKEN', None),
               Option('napps', 'api', 'NAPPS_API_URI', 'https://napps.kytos.io/api/'),
               Option('napps', 'repo', 'NAPPS_REPO_URI', 'https://napps.kytos.io/repo'),
               Option('kytos', 'api', 'KYTOS_API', 'http://localhost:8181')]

for option in options:
    env_value = os.environ.get(option.env_var)
    if env_value:
        self.config.set(option.section, option.name, env_value)
    elif option.default_value:
        self.config.set(option.section, option.name, option.default_value)
```
This avoids code repetition and therefore mistakes, while facilitation the addition of new parameters.

File: kytos/utils/napps.py [hunk 2]
```diff
@@ -36,12 +36,24 @@ def __init__(self, controller=None):
...
+        """Request current configurations loaded by Kytos instance."""
+        uri = self._kytos_api + 'kytos/configuration/'
+        try:
+            options = json.loads(urllib.request.urlopen(uri).read())
+        except urllib.error.URLError:
+            print('Kytos is not running.')
+            sys.exit()
+
+        self._installed = Path(options.get('installed_napps'))
+        self._enabled = Path(options.get('napps'))
```
Comment (diraol) [inline kytos/utils/napps.py]: is it just `napps` (the path for the enabled napps)?

Comment (macartur) [inline kytos/utils/napps.py]: Yes, it is. The controller.options have a field name 'napps' that store the Napps directory.

**Comment by macartur** [comment]
I found a problem in this pull request, I will fix this and refresh this pull request.

File: kytos/utils/config.py [hunk 3]
```diff
@@ -58,46 +59,33 @@ def set_env_or_defaults(self):
...
+                          'https://napps.kytos.io/api/'),
+                   Option('napps', 'repo', 'NAPPS_REPO_URI',
+                          'https://napps.kytos.io/repo'),
+                   Option('kytos', 'api', 'KYTOS_API',
+                          'http://localhost:8181/')]
+
+        for option in options:
+            if not self.config.has_option(option.section, option.name):
+                env_value = os.environ.get(option.env_var)
+                if env_value and not hasattr(self.config, options):
```
Comment (diraol) [inline kytos/utils/config.py]: `hasattr(self.config, options)` does not make sense ....

File: kytos/utils/config.py [hunk 4]
```diff
@@ -58,46 +59,33 @@ def set_env_or_defaults(self):
...
+                    self.config.set(option.section, option.name, env_value)
+                elif option.default_value:
+                    self.config.set(option.section, option.name,
+                                    option.default_value)
```
Comment (diraol) [inline kytos/utils/config.py]: Also, I have just thought that we can reduce the code here with:
```python
for option in options:
    if not self.config.has_option(option.section, option.name):
        env_value = os.environ.get(option.env_var, option.default_value)
        if env_value and not hasattr(self.config, options):
            self.config.set(option.section, option.name, env_value)
```
(Consider that I haven't understood the `hasattr(self.config, options)` condition....)

File: kytos/utils/config.py (hunk 3 above)
Comment (macartur) [inline kytos/utils/config.py]: You are right, I will remove this.

File: kytos/utils/config.py (hunk 4 above)
Comment (macartur) [inline kytos/utils/config.py]: Done.

File: CHANGELOG.rst [hunk 5]
```diff
@@ -10,6 +10,8 @@ Added
 
 Changed
 =======
+- Install and enable NApps based on Kytos instance. kytos commands will request
+  the configuration loaded by kytos before install and enable new napps.
```
Comment (cemsbr) [inline CHANGELOG.rst]: kytos commands -> kytos-utils
install and enable new napps -> managing NApps

Comment (macartur) [inline CHANGELOG.rst]: Done.

**Comment by beraldoleal** [comment]
@cemsbr your changes are ok ?

**Pull request merged**